subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptions.java

```diff
@@ -168,8 +168,9 @@ public void applyTo(StartParameter settings, Origin origin) {
     }
 
     public static class NoProjectDependenciesRebuildOption extends EnabledOnlyBooleanBuildOption<StartParameter> {
+        public static final String DEPRECATION_MESSAGE = "Support for --no-rebuild and -a was deprecated and is scheduled to be removed in Gradle 5.0.";
```

Comment:
:x: I think this has come up in a different review as well. The error message should be constructed through some other logic to make it easier and keep it unified. We could pull out some logic from `SingleMessageLogger`.

Reply:
Good advice. Let's make it at another PR so that the review can be easier.

Reply:
Sounds good to me.